### DISCUSSION TITLE
StarRocks Internal Table Format advantages over Apache Iceberg

### DISCUSSION THREAD

**Discussion by alberttwong** [StarRocks/starrocks#34360] [How-tos, Tutorials, Best Practices and Architecture Articles]
The StarRocks internal format will be better than query on Iceberg. The internal format has more optimization, like late materialization, partition pruning, more predicates pushed down.

Using Shared Nothing Architecture, this StarRocks Table format is stored on disk.  Using Shared Data architecture, this StarRocks Table format is stored in S3.

**Comment by mcgray** [comment]
Can you elaborate, please? What is this internal format specifically?

**Reply by atifiu** [reply]
Does SR supports hidden partitioning like iceberg ? Is there any videos on such features of SR?

**Reply by alberttwong** [reply]
It is our own format.  Just like big query or redshift has their own format.   It does have partitioning.   All the instructions are on our format unless it says it's Apache Iceberg or Apache Hudi or one of the other table formats.

**Reply by thevijayshekhawat** [reply]
Where can I read more about this Internal Storage format? Can you please share some links to the documentation?

**Reply by thevijayshekhawat** [reply]
Also, can this storage format be used to store data in Object storage like S3 and GCS and query directly from Starrocks?

**Reply by atifiu** [reply]
Internal format means SR proprietary format like oracle or sql server.

**Reply by alberttwong** [reply]
I updated the answer to include object store.

**Reply by thevijayshekhawat** [reply]
Thanks, @alberttwong , Follow-up question, is this format available with StarRocks 3.1? Also, Can you share some links to documentation about how this internal table format can be used to create tables on Object Store? We are currently testing StarRocks with IceBerg and it would be great to compare the performance across different table formats.

**Reply by alberttwong** [reply]
This is the default when you create "data" in starrocks. It's listed as "internal catalog".

**Comment by alberttwong** [comment]
moved to https://forum.starrocks.io/t/starrocks-internal-table-format-advantages-over-apache-iceberg/121
